Bricks/mq/inmemory/test.cc

```diff
@@ -86,26 +91,31 @@ TEST(InMemoryMQ, DropOnOverflowTest) {
   for (size_t i = 0; i < 20; ++i) {
     mmq.PushMessage(bricks::strings::Printf("M%02d", static_cast<int>(i)));
   }
-  // Wait until at least three messages are processed by the consumer to
-  // properly get dropped messages count.
-  while (c.processed_messages_ < 2) {
-    ;  // Spin lock;
-  }
 
-  // Confirm that exactly 10 messages were dropped due to a very slow consumer.
+  // Confirm that exactly 10 messages were processed and exactly 10 were dropped due to a very slow consumer.
   // Despite being flaky, this condition can only fail if it takes more than 20 ms to add the messages,
   // which is quite huge time interval.
-  EXPECT_EQ(c.dropped_messages_, 10u);
-
-  // Wait until the rest of the queued messages are processed.
   c.delay_ms_ = 0u;
   while (c.processed_messages_ != 10u) {
-    ;  // Spin lock;
+    ;  // Spin lock.
   }
 
+  EXPECT_FALSE(c.at_least_one_message_dropped_);
```

Comment:
Yes, I knew it looks strange, and thanks for asking the right question.

I thought of it a lot and realized it's easier for both the user and the for the interface if the fact that the message is being dropped is immediately returned to the code that is synchronously pushing the entry, not to the that is asynchronously processing them.

Reworded the comment and the test in 5845d14. PTAL.